storagemarket/impl/providerstates/provider_fsm.go

```diff
@@ -262,3 +262,13 @@ var ProviderFinalityStates = []fsm.StateKey{
 	storagemarket.StorageDealSlashed,
 	storagemarket.StorageDealExpired,
 }
+
+// ProviderSealingStates are the states on the happy path after hand-off to
+// the sealing subsystem
+var ProviderSealingStates = []fsm.StateKey{
```

Comment:
Are we 100% sure that no other deals need to be registered in the DAG store?

Reply:
I'm not sure I understand the question - are you asking about which are the states that deals requiring registration are in?

Basically it's any state after and including `StorageDealStaged` - This is the state in the new code in which `RegisterShard` is called.

The termination states are
- `storagemarket.StorageDealError`
- `storagemarket.StorageDealSlashed`
- `storagemarket.StorageDealExpired`
We don't want to register deals in any of these states.

So the remaining possibilities are:
- `storagemarket.StorageDealStaged`
- `storagemarket.StorageDealAwaitingPreCommit`
- `storagemarket.StorageDealSealing`
- `storagemarket.StorageDealFinalizing`
- `storagemarket.StorageDealActive`

You can examine [the state diagram](https://github.com/filecoin-project/go-fil-markets/blob/master/docs/storageprovider.mmd.png) to make sure.

Reply:
Thanks, that makes sense!